File: javapatterns/src/main/java/org/iut/nantes/adapter/canards/Dinde.java
package org.iut.nantes.adapter.canards;

public interface Dinde {
    void voler();
    void glousser();
}

class DindeSauvage implements Dinde {

    @Override
    public void voler() {
        System.out.println("Je peux voler, mais j'ai la flemme.");
    }

    @Override
    public void glousser() {
        System.out.println("Glou glou glou.");
    }
}

File: kotlinpatterns/src/main/kotlin/org/iut/nantes/adapter/canards/Dinde.kt
package org.iut.nantes.adapter.canards

interface Dinde {
    fun voler()
    fun glousser()
}

internal class DindeSauvage : Dinde {
    override fun voler() {
        println("Je peux voler, mais j'ai la flemme.")
    }

    override fun glousser() {
        println("Glou glou glou.")
    }
}